components/brave_today/browser/urls.cc

```diff
@@ -25,16 +36,36 @@ std::string GetHostname() {
   }
 }
 
-std::string GetRegionUrlPart() {
+std::string GetV1RegionUrlPart() {
   const std::string locale =
       brave_l10n::LocaleHelper::GetInstance()->GetLocale();
   const std::string language_code = brave_l10n::GetLanguageCode(locale);
-  // TODO(petemill): Have a remotely-updatable list of supported language
-  // variations.
   if (language_code == "ja") {
     return "ja";
   }
   return "";
 }
 
+std::string GetRegionUrlPart() {
+  if (!base::FeatureList::IsEnabled(
+          brave_today::features::kBraveNewsV2Feature)) {
+    return GetV1RegionUrlPart();
+  }
+
+  // Locale is lang-COUNTRY but Brave News wants the format to be lang_COUNTRY.
+  const std::string locale =
+      brave_l10n::LocaleHelper::GetInstance()->GetLocale();
+  const std::string language_code = brave_l10n::GetLanguageCode(locale);
+  const std::string country_code = brave_l10n::GetCountryCode(locale);
+  const std::string brave_news_locale = language_code + "_" + country_code;
+
+  if (base::Contains(kSupportedLocales, brave_news_locale))
+    return brave_news_locale;
+
+  // Fallback to en_US, if we can't match anything else.
+  // TODO(fallaciousreasoning): Come up with some more complicated fallback
+  // mechanism.
+  return "en_US";
```

Comment:
Reminder to tweak this based on what we've written in the internationalization spec doc (under "Phase 1")

Reply:
Hey, I'd like to land this more or less "as is" for now. I think we can add the more complicated fallbacks pretty easily later, without risking breaking anything. WDYT?

Reply:
(so I can find it, here's the internationalization doc: https://docs.google.com/document/d/1O1il9DW3Z5b0Twb3x37Ar2g48Zhszr7NXzsM_VK-aCQ/edit#heading=h.hwillwfhgsqb)